mw-webapp/src/pages/userPage/UserPage.tsx

```diff
@@ -0,0 +1,59 @@
+import {useEffect, useState} from "react";
+import {User} from "firebase/auth";
+import {ReportsTable} from "src/pages/reportsTable/ReportsTable";
+import {handleUserAuthState} from "src/service/auth/handleUserAuthState";
+import {UserService} from "src/service/UserService";
+
+/**
+ * Default page for authorized User
+ */
+export const UserPage = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [ways, setWays] = useState<string[] | null>(null);
+
+  useEffect(() => {
+    handleUserAuthState(setUser);
+  }, []);
+
+  /**
+   * Get owned Ways of authorized user
+   */
+  const getUserWays = async (uuid: string) => {
+    const foundUser = await UserService.getUserDTO(uuid);
+    setWays(foundUser.ownWays);
+  };
+
+  useEffect(() => {
+    if (user) {
+      getUserWays(user.uid);
+    } else {
+      setWays(null);
+    }
+  }, [user]);
+
+  return (
+    <>
+      {ways
+        ? (
+          ways.map((wayUuid) => (
+            <>
+              <h4>
+                Table for
+                &quot;
+                {wayUuid}
+                &quot;
+                way
+              </h4>
+              <ReportsTable
+                wayUuid={wayUuid}
+                key={wayUuid}
+              />
+            </>
+          ))
+        )
+        : (
+          null
+        )}
+    </>
```

Comment:
DRY. this fragment practically repeats 1 in 1 as in [AllWaysPage.tsx](https://github.com/tritonJS826/masters-way/pull/139/files#diff-48f651b7b1f7b206239eed9e44cde852d68a5151847554d25272dc381652e501)

Reply:
Refactored